### PR TITLE
refactor: add zone state view

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes https://github.com/cheminfo/nmrium/issues/1722

### CODE REVIEW

File: src/component/panels/ZonesPanel/ZonesPanel.tsx [hunk 1]
```diff
@@ -154,9 +170,15 @@ function ZonesPanelInner({ zones, activeTab, xDomain, yDomain, experiment }) {
     setFlipStatus(false);
   }, []);
 
-  const visibilityHandler = useCallback((key) => {
-    Events.emit('onZonesVisibilityChange', { key });
-  }, []);
+  const handleSetShowZones = useCallback(() => {
```
Comment (stropitek) [inline src/component/panels/ZonesPanel/ZonesPanel.tsx:173]: using useCallback or useMemo for things that are passed to a component that is not memoized is actually a (very small) decrease in performance

In general our approach to performance is do not worry about it too much at first unless there are heavy computations involved, or you are working with state (typically in a react context) which trigger the re-rendering of the whole page.

At some point we will refocus on performance looking at bottlenecks, but it is not a priority today.

File: src/component/panels/ZonesPanel/ZonesPanel.tsx [hunk 2]
```diff
@@ -258,18 +280,26 @@ const MemoizedZonesPanel = memo(ZonesPanelInner);
...
-
+  const {
+    displayerKey,
+    xDomain,
+    yDomain,
+    activeTab,
+    view: { zones: zoneState },
+  } = useChartData();
+  const { zones, info, id } = useSpectrum(emptyData) as Datum2D;
+  const btnState = zoneState.find((r) => r.spectrumID === id) || zoneStateInit;
```
Comment (stropitek) [inline src/component/panels/ZonesPanel/ZonesPanel.tsx:291]: why `btnState`? and not something like `zoneProps`?

File: src/component/panels/ZonesPanel/ZonesPanel.tsx (hunk 1 above)
Comment (wadjih-bencheikh18) [reply src/component/panels/ZonesPanel/ZonesPanel.tsx:173]: Do you want me to revert states changes ?

Comment (stropitek) [reply src/component/panels/ZonesPanel/ZonesPanel.tsx:173]: No, just remove the 3 next `useCallback` as you can just declare regular functions without hurting performance

Comment (stropitek) [reply src/component/panels/ZonesPanel/ZonesPanel.tsx:173]: I invite you to read
https://www.joshwcomeau.com/react/why-react-re-renders/
https://www.joshwcomeau.com/react/usememo-and-usecallback/

Comment (wadjih-bencheikh18) [reply src/component/panels/ZonesPanel/ZonesPanel.tsx]: > just remove the 3 next `useCallback`

Done,fixed


> I invite you to read

Thank you

**Review by stropitek — APPROVED** [review]
LGTM after conflicts are resolved

**Pull request merged**